src/core/attribute.d

```diff
@@ -51,7 +63,67 @@ module core.attribute;
  * }
  * ---
  */
-version (D_ObjectiveC) struct selector
+version (UdaSelector) struct selector
 {
     string selector;
 }
+
+/**
+ * Use this attribute to declare an ABI tag on a C++ symbol.
+ *
+ * ABI tag is an attribute introduced by the GNU C++ compiler.
+ * It modifies the mangled name of the symbol to incorporate the tag name,
+ * in order to distinguish from an earlier version with a different ABI.
+ * See: $(LINK2 https://gcc.gnu.org/onlinedocs/gcc/C_002b_002b-Attributes.html, GCC attributes documentation).
+ *
+ * This is a special compiler recognized attribute, it has a few
+ * requirements, which all will be enforced by the compiler:
+ *
+ * $(UL
+ *  $(LI
+ *      The attribute can only be attached to a declaration of
+ *      an $(D_CODE extern(C++)) variable, function, aggregate,
+ *      enum, template, or namespace.
+ *  ),
+ *
+ *  $(LI
+ *      The string arguments must only contain valid characters
+ *      for C++ name mangling which currently include alphanumerics
+ *      and the underscore character.
+ *  ),
+ * )
+ *
+ * Note: Unlike other attributes, if applied to a namespace,
+ * this attribute it is not inherited by the members.
```

Comment:
>  Tags on an inline namespace are not represented in the mangled name of the namespace, but they are subject to the above tag propagation

https://itanium-cxx-abi.github.io/cxx-abi/abi.html#mangle.abi-tag

> The attribute can also be applied to an inline namespace, but does not affect the mangled name of the namespace; in this case it is only used for -Wabi-tag warnings and automatic tagging of functions and variables. Tagging inline namespaces is generally preferable to tagging individual declarations, but the latter is sometimes necessary, such as when only certain members of a class need to be tagged.

https://gcc.gnu.org/onlinedocs/gcc/C_002b_002b-Attributes.html

So it should be inherited by the member. But that's a C++ side thing, we should represent this with `@gnuAbiTag("XXX") { /* decls */ }` not by trying to mimic the C++ code.
I would just remove this sentence.

Reply:
I don't understand what you mean.
That sentence is about the current behavior of DMD of inheriting UDAs for a namespace.
for example:
```d
@("abc")
extern (C++, "namespace")
{
    void fun();
}
```
`fun` has the attribute `"abc"`.

while it is not the same in the following case:
```d
@("abc")
extern (C++)
class C
{
    void fun();
}
```
`fun` doesn't have the attribute `"abc"`.

Reply:
This special UDA is not supposed to be inherited from a namespace.

Reply:
Yes and that makes sense. But what your sentence reads to me is that in this case:
```d
@gnuAbiTag("Test")
extern (C++, "Test") {
    void fun();
}
```
`fun` would not have the `funAbiTag("Test")` attribute.

Reply:
Yes that's what I meant, It is handled specially by the compiler that's why it's documented.